src/components/Date.jsx

```diff
@@ -167,6 +167,13 @@ const MyStartDateSelector = ({
     <>
       <DatePicker
         selected={startDate}
+        /**
+         * This onChange method only runs when the date picker component is renderen (of course)
+         * This means that a user can pick a bad date, re-render the component and confirm the bad date 
+         * since the date check only runs on change (when the user clicks another date) and the date 
+         * the user chose last is saved even if the date picker is re-rendered
+         * To fix this: Reset date every time the date picker is re-rendered
+         */
```

Comment:
Retarded explanation of the problem, but I hope you get the point

Reply:
So it needs a clear/clan-up function? Or am I misunderstanding? But maybe hook the datepicker component up with the map first - there might be more bugs after that.

Reply:
Yes I think saving the initial dates that are shown when the component is rendered first time would help solve the problem - then we can set the date to the initial value every time the component re-renders. Solving the problem this way also means that we don't have to change anything in the onChange-method. 
I don't think we have to wait till the date picker has been hooked up with the map. But there's also no rush and you're right - maybe more bugs will appear which could be included in this bug-fix thread.